// @leonardo-ai/sdk/sdk/sdk.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Leonardo = void 0;
const sdks_js_1 = require("../lib/sdks.js");
const dataset_js_1 = require("./dataset.js");
const elements_js_1 = require("./elements.js");
const image_js_1 = require("./image.js");
const initimages_js_1 = require("./initimages.js");
const models_js_1 = require("./models.js");
const motion_js_1 = require("./motion.js");
const pricingcalculator_js_1 = require("./pricingcalculator.js");
const prompt_js_1 = require("./prompt.js");
const realtimecanvas_js_1 = require("./realtimecanvas.js");
const texture_js_1 = require("./texture.js");
const threedmodelassets_js_1 = require("./threedmodelassets.js");
const user_js_1 = require("./user.js");
const variation_js_1 = require("./variation.js");
class Leonardo extends sdks_js_1.ClientSDK {
    get initImages() {
        var _a;
        return ((_a = this._initImages) !== null && _a !== void 0 ? _a : (this._initImages = new initimages_js_1.InitImages(this._options)));
    }
    get dataset() {
        var _a;
        return ((_a = this._dataset) !== null && _a !== void 0 ? _a : (this._dataset = new dataset_js_1.Dataset(this._options)));
    }
    get elements() {
        var _a;
        return ((_a = this._elements) !== null && _a !== void 0 ? _a : (this._elements = new elements_js_1.Elements(this._options)));
    }
    get image() {
        var _a;
        return ((_a = this._image) !== null && _a !== void 0 ? _a : (this._image = new image_js_1.Image(this._options)));
    }
    get realtimeCanvas() {
        var _a;
        return ((_a = this._realtimeCanvas) !== null && _a !== void 0 ? _a : (this._realtimeCanvas = new realtimecanvas_js_1.RealtimeCanvas(this._options)));
    }
    get motion() {
        var _a;
        return ((_a = this._motion) !== null && _a !== void 0 ? _a : (this._motion = new motion_js_1.Motion(this._options)));
    }
    get texture() {
        var _a;
        return ((_a = this._texture) !== null && _a !== void 0 ? _a : (this._texture = new texture_js_1.Texture(this._options)));
    }
    get user() {
        var _a;
        return ((_a = this._user) !== null && _a !== void 0 ? _a : (this._user = new user_js_1.User(this._options)));
    }
    get models() {
        var _a;
        return ((_a = this._models) !== null && _a !== void 0 ? _a : (this._models = new models_js_1.Models(this._options)));
    }
    get threeDModelAssets() {
        var _a;
        return ((_a = this._threeDModelAssets) !== null && _a !== void 0 ? _a : (this._threeDModelAssets = new threedmodelassets_js_1.ThreeDModelAssets(this._options)));
    }
    get pricingCalculator() {
        var _a;
        return ((_a = this._pricingCalculator) !== null && _a !== void 0 ? _a : (this._pricingCalculator = new pricingcalculator_js_1.PricingCalculator(this._options)));
    }
    get prompt() {
        var _a;
        return ((_a = this._prompt) !== null && _a !== void 0 ? _a : (this._prompt = new prompt_js_1.Prompt(this._options)));
    }
    get variation() {
        var _a;
        return ((_a = this._variation) !== null && _a !== void 0 ? _a : (this._variation = new variation_js_1.Variation(this._options)));
    }
}
exports.Leonardo = Leonardo;
